jiant/tasks/tasks.py

```diff
@@ -2946,6 +3005,66 @@ class SpanPredictionTask(Task):
 
     n_classes = 2
 
+    def update_metrics(self, out, batch):
+        batch_size = sum(out["n_exs"]).item()
```

Comment:
This line breaks `SpanPredictionTask`s (i.e., `QASRL` and `QAMR`) in single-GPU mode:

```
Traceback (most recent call last):
  File "/home/pcy214/jiant/main.py", line 27, in <module>
    raise e  # re-raise exception, in case debugger is attached.
  File "/home/pcy214/jiant/main.py", line 16, in <module>
    main(sys.argv[1:])
  File "/home/pcy214/jiant/jiant/__main__.py", line 638, in main
    phase="target_train",
  File "/home/pcy214/jiant/jiant/trainer.py", line 579, in train
    output_dict = self._forward(batch, task=task)
  File "/home/pcy214/jiant/jiant/trainer.py", line 1035, in _forward
    task.update_metrics(model_out, batch)
  File "/home/pcy214/jiant/jiant/tasks/tasks.py", line 3009, in update_metrics
    batch_size = sum(out["n_exs"]).item()
TypeError: 'int' object is not iterable
```

Looks like a minimal fix could just check the type here and handle `int`s (for single-GPU mode).

Reply:
cuda_device isn't passed in here, so I think it might be a bit more involved. Instead, I'll check for batch size from somewhere else in out.